light/store/db/db.go

```diff
@@ -147,13 +147,8 @@ func (s *dbs) LastLightBlockHeight() (int64, error) {
 	}
 	defer itr.Close()
 
-	for itr.Valid() {
-		key := itr.Key()
-		_, height, ok := parseLbKey(key)
-		if ok {
-			return height, nil
-		}
-		itr.Next()
+	if itr.Valid() {
+		return parseHeightFromKey(itr.Key()), nil
```

Comment:
are u saying we no longer need to skip records in order to find key? that's cool

Reply:
Hmm I was actually just deducing this from the current logic and thought I could make it simpler. I'm second guessing myself now haha so I'll quickly write some tests to confirm that it works as expected. I believe it is the iterators job to find the array of keys based of the start and end bytes given as arguments. I.e. if `s.lbKey(1)` isn't an actual key in the db it will start at the next (lexicographically-ordered) one that exists which should be the key of the first height.